### ISSUE TITLE
storeSubscription context object is lost when a component don't have a mapStateToProps

### ISSUE THREAD

**Issue by ericm546** [reactjs/react-redux#577]
Hi, 

If I have the following components: 

```javascript
const Child = connect(childMapState, childMapDispatch, () => <div />);
const Middle = connect(null, middleMapDispatch, () => <Child />);
const Parent = connect(parentMapState, parentMapDispatch, () => <Middle />);
```

Because, the middle component has no mapStateToProps, the value of shouldHandleStateChanges will be false in connectAdvanced and getChildContext() will put undefined as the storeSubscription value sent to the Child component.

This cause my application to have the following warning, when an action cause some components to be unmounted: 
`Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the Connect(Child) component.`

I was not able to reproduce it in a separated project, but it fix the problem if I either put an "if" inside the  getChildContext function of the connectAdvanced component: 
```javascript
if(this.subscription ) {
    return {[subscriptionKey]: this.subscription}
} else {
  return {}
}
```
or if I put a function returning an empty object as the first parameter to connect.

Note that I tested it in react-redux 5.0 and 5.0.1 and this was working before in 4.X.

**Comment by jimbolla** [comment]
Good catch. It should actually be
```
getChildContext() {
  return { [subscriptionKey]: this.subscription || this.parentSub }
}
```
I think. I'll have to test this tomorrow.

**Comment by markerikson** [comment]
Awww.  Why couldn't people find these _before_ we released 5.0.0 final?  :(

Thanks for the report, @ericm546 .  Appreciate it!

**Comment by ericm546** [comment]
jimbolla: That fixes the bug on my project. Thank you for being so quick :)
